File: lib_base/src/main/java/com/android/base/app/ui/RefreshListLayout.java
package com.android.base.app.ui;

import java.util.List;

/**
 * 列表视图行为
 *
 * @author Ztiany
 * Email: 1169654504@qq.com
 * Date : 2017-03-29 22:16
 */
public interface RefreshListLayout<T> extends RefreshStateLayout {

    void replaceData(List<T> data);

    void addData(List<T> data);

    PageNumber getPager();

    boolean isEmpty();

    void loadMoreCompleted(boolean hasMore);

    void loadMoreFailed();

    boolean isLoadingMore();

    @Override
    boolean isRefreshing();

}

File: lib_base/src/main/java/com/android/base/architecture/ui/RefreshListLayout.kt
package com.android.base.architecture.ui

/**
 * 列表视图行为
 *
 * @author Ztiany
 * Email: 1169654504@qq.com
 * Date : 2017-03-29 22:16
 */
interface RefreshListLayout<T> : RefreshStateLayout {

    fun replaceData(data: List<T>)

    fun addData(data: List<T>)

    fun loadMoreCompleted(hasMore: Boolean)

    fun loadMoreFailed()

    fun getPager(): Paging

    fun isEmpty(): Boolean

    fun isLoadingMore(): Boolean

    fun isLoadMoreEnable(): Boolean

}